pkg/validation/internal/bundle.go

```diff
@@ -133,23 +134,66 @@ func getOwnedCustomResourceDefintionKeys(csv *operatorsv1alpha1.ClusterServiceVe
 // - we could identify that the bundle size is close to the limit (bigger than 85%)
 func validateBundleSize(bundle *manifests.Bundle) []errors.Error {
 	warnPercent := 0.85
-	warnSize := int64(max_bundle_size * warnPercent)
+	warnSize := float64(max_bundle_size) * warnPercent
 	var errs []errors.Error
 
 	if bundle.CompressedSize == nil || *bundle.CompressedSize == 0 {
-		errs = append(errs, errors.WarnFailedValidation("unable to check the bundle size", nil))
+		errs = append(errs, errors.WarnFailedValidation("unable to check the bundle compressed size", bundle.Name))
 		return errs
 	}
 
+	if bundle.Size == nil || *bundle.Size == 0 {
+		errs = append(errs, errors.WarnFailedValidation("unable to check the bundle size", bundle.Name))
+		return errs
+	}
+
+	// From OPM (https://github.com/operator-framework/operator-registry) 1.17.5
+	// and OLM (https://github.com/operator-framework/operator-lifecycle-manager) : v0.19.0
+	// the total size checked is compressed
 	if *bundle.CompressedSize > max_bundle_size {
-		errs = append(errs, errors.ErrInvalidBundle(fmt.Sprintf("maximum bundle compressed size with gzip size exceeded: size=~%d MegaByte, max=%d MegaByte", *bundle.CompressedSize/(1<<(10*2)), max_bundle_size/(1<<(10*2))), nil))
-	} else if *bundle.CompressedSize > warnSize {
-		errs = append(errs, errors.WarnInvalidBundle(fmt.Sprintf("nearing maximum bundle compressed size with gzip: size=~%d MegaByte, max=%d MegaByte", *bundle.CompressedSize/(1<<(10*2)), max_bundle_size/(1<<(10*2))), nil))
+		errs = append(errs, errors.ErrInvalidBundle(
+			fmt.Sprintf("maximum bundle compressed size with gzip size exceeded: size=~%s , max=%s. Bundle uncompressed size is %s",
+				formatBytesInUnit(*bundle.CompressedSize),
+				formatBytesInUnit(max_bundle_size),
+				formatBytesInUnit(*bundle.Size)),
+			bundle.Name))
+	} else if float64(*bundle.CompressedSize) > warnSize {
+		errs = append(errs, errors.WarnInvalidBundle(
+			fmt.Sprintf("nearing maximum bundle compressed size with gzip: size=~%s , max=%s. Bundle uncompressed size is %s",
+				formatBytesInUnit(*bundle.CompressedSize),
+				formatBytesInUnit(max_bundle_size),
+				formatBytesInUnit(*bundle.Size)),
+			bundle.Name))
+	}
+
+	// Before these versions the bundles were not compressed
+	// and their size must be < ~1MB
```

Comment:
I feel somewhat strongly that we shold keep the API repo focused on just one level of compatibility with OPM/OLM, so I'd suggest we don't check the uncompressed size at all since current versions of OPM/OLM do compression.

Otherwise, it seems like the API repo's main branch will start growing to encompass validation for all past history of what OPM/OLM support.

Reply:
I will remove it and add in another PR so we are able to discuss it.
oo, we can move with the fix aasp.